### DISCUSSION TITLE
Rationalizing Counterintuitive Shap values

### DISCUSSION THREAD

**Discussion by datascienceenthu1** [shap/shap#2997] [General]
Suppose we train a single binary classification model on tabular data with 3 cohorts C1, C2, C3 with 100 features F1 to F100.  We observe a certain distribution of features Fis in cohort C1 and using other metrics we expect a certain ranking of features for any record from C1. 

But when we look at Shap values- sometimes the Shap rank or Shap value tends to be counterintuitive.. Say F10 which is not expected to have much predictive power in Cohort C1 (given either range of F10 is more narrow in C1 or observed distribution of F10 is as such in C1 ) has a high Shap rank /value for a record from C1.  

 F10 ends up having a high Shap value probably because while evaluating marginal contribution- multiple values of F10 are tried outside typical range in C1.  How do we explain this and how do we prevent this ?
If we use a background dataset is this expected ?

**Comment by jainraj** [comment]
Would it be possible to see some plots - maybe helpful to understand the problem in detail. 

From what I understand these two papers might be relevant: 

1. [The Explanation Game: Explaining Machine Learning Models Using Shapley Values - Merrick and Taly](https://arxiv.org/pdf/1909.08128.pdf)
2. [True to the Model or True to the Data? - Chen et al.](https://arxiv.org/pdf/2006.16234.pdf)
